webroot/adm/keysets.html

```diff
@@ -69,6 +91,7 @@ <h3>Output</h3>
     $('#doUpdate').on('click', function () {
       const siteId = parseInt($('#siteId').val());
```

Comment:
maybe out of scope for this ticket, just noticed this when i'm testing - what is the purpose of "Update Keyset"?  
so it can do 

1. update the Site Id this keyset is for
2. Update the keyset name
3 Add/remove allow site ids

There are a few issues:

1. Turns out Name/Add/Remove SiteId fields are optional but Site Id is compulsory even if we are not changing it. Is this really necessary if Keyset id is always required and we aren't changing site id? 
2. If Keyset Id is missing, it just said Error: 400: Bad Request. Could we improve it so it said Keyset Id is missing and that's a must have?
3. there is no check on the validity of site ids on Site Id (i can put random number into associated Site Id and the update will be successful), Add SiteIds and Remove Site Ids. Should there be otherwise people can mistypes and mistakes are ignored?

I can't remember if site id values arechecked before keyset work.


@jon8787 @cody-constine-ttd @zaiweidu thoughts?

Reply:
1.  So "update keyset" uses the same endpoint as "make new keyset", with the siteid being used to create a new keyset if no keyset id is provided. But if keyset id is provided then site id is not needed. Should we even allow an existing keyset to have it's site id reassigned to a different site? Versus just make new keyset for the other site id?
2. Putting in a check on the ui side for if keysetId/siteId is not there, as otherwise it ends up trying to call it with "NaN". I also updated the backend to return more helpful message. 
3. I can't see any reason not to do this. I have added this check in and updated tests in SharingServiceTest.

Reply:
For 1., will let you update the code from slack chat result first.
For 2., the update looks ok
For 3, is there a check on _Add Site id_'s value? I _allowed_ site id 33 as shown in screencap which doesn't exist but it still allows me adding it and shown in the output? 
![image](https://github.com/IABTechLab/uid2-admin/assets/2499852/0ee89c0c-5484-4bd3-a775-f844b88636ae)

And currently if the Site ID field is set to invalid site id and i click update, there's only a error logged in admin service log like:
_ERROR com.uid2.admin.vertx.ResponseUtil - Site id 999 not valid_

but nothing shown on the html web page Output section? Probably wanna check when the API call return errors, whether those errors will appear on the webpage output

Reply:
@Ian-Nara for #1, we do NOT allow Keyset's siteId change once it's created.

Reply:
Right now it just checks the keyset owner's site to be valid. I will add allowlist also, but am moving this to another branch

Reply:
Also fixed error output not showing for update/reset